### DISCUSSION TITLE
Help me understand energy dahsboard

### DISCUSSION THREAD

**Discussion by SKNBP** [wlcrs/huawei_solar#716] [General]
Can't understand whats is going on on energy dashboard. 
Why i exports negative energy during night(as shown in purple on screenshot). Does it count to "Net returned to the grid" also as purple part? 
Could someone helps me understand how can i read those numbers and this dashboard?

Some screenshots:
![energy-dashboard-settings-consumption](https://github.com/wlcrs/huawei_solar/assets/105205644/2aa43082-f356-4117-aee4-14639df12630)
![energy-dashboard-settings-export](https://github.com/wlcrs/huawei_solar/assets/105205644/606d003f-fb6b-456b-8b40-eb937b3dc147)
![dashboard-few-days](https://github.com/wlcrs/huawei_solar/assets/105205644/f590b159-48f0-4792-bafc-4a869937c522)
![dashboard-one-day](https://github.com/wlcrs/huawei_solar/assets/105205644/9ada017a-0da3-4f76-a9f5-bd2577b7056a)

**Comment by Roving-Ronin** [comment]
@SKNBP including a description of your setup might be handy...
That said (even though in foreign language) it appears that your using the Power Meter's actual direct sensors in the Energy Dashboard.... this would mean your going to get a huge number when you add it to a HA instance (as its the meters LIFETIME reading, not that days import/export).

Suggest you look at creatinng a sensor for the consumption of the house and utility_meter (or better use the integration for energy_meter, that add $ in/out tracking based of a sensor of the current price). You then select the Daily kWh Exported or Imported utility meter sensors to use in the Energy Dashboard.

This is for another project for the Power Flow (SunSynk card) but see for some ideas: https://github.com/Roving-Ronin/sunsynk-power-flow-card/blob/main/docs/examples/huawei_packages/huawei_derived_sensors.yaml

```
  # Provides the calculated current load, in Watts, of the house including power being drawn from solar, battery & grid.
  - sensor:
    - name: "House Consumption - Power"
      unique_id: house_consumption_power
      unit_of_measurement: "W"
      icon: mdi:home-lightning-bolt-outline
      device_class: "power"
      state_class: measurement
      state: >-
        {% set inverters_power = states('sensor.inverters_active_power') | float(default=0) %}
        {% set power_meter_power = states('sensor.power_meter_active_power') | float(default=0) %}

        {{ max(0, (inverters_power - power_meter_power)) }}

      availability: >-
        {{  states('sensor.inverters_active_power') | is_number
        and states('sensor.power_meter_active_power') | is_number }}
```

and

```
########### Start: Helpers - Riemann Sum to conver Power (W) to Energy (kWh) ###########

  # Riemann Sum sensor to take "Inverters - Input Power with efficiency loss" and convert it into an ENERGY sensor.
sensor:
  - platform: integration
    name: "Inverter - Input Power - Energy"
    unique_id: inverters_input_power_energy
    source: sensor.inverters_input_power_with_efficiency_loss
    round: 3
    unit_prefix: k
    unit_time: h
    method: left


  # Riemann Sum sensor to take "House Consumption - Power" and convert it into ENERGY sensor.
  - platform: integration
    name: "House Consumption - Energy"
    unique_id: house_consumption_energy
    source: sensor.house_consumption_power
    round: 3
    unit_prefix: k
    unit_time: h
    method: left


########### Start: Energy Meter - Used to create sensors that track usage over a set period ###########
#
#  Note: Energy_Meter a HACS integration is used INSTEAD OF the utility_meter, with it providing 
#        additional functionality, including the ability to quickly add costs to energy sensors.
#  See:  https://github.com/zeronounours/HA-custom-component-energy-meter
#
energy_meter:

  # Used to convert the above Riemann sum sensor into an ENERGY sensor that RESETS to ZERO DAILY. 
  house_consumption_energy_daily:
    unique_id: house_consumption_energy_daily
    name: "House Consumption - Energy - Daily"
    source: sensor.house_consumption_energy
    cycle: daily

######################### End of File #########################    
```

For an idea re the utility_meter / energy_meter and defining sensors for the currect cost to import, or paid to export see:

https://github.com/Roving-Ronin/sunsynk-power-flow-card/blob/main/docs/examples/huawei_packages/huawei_solar_electricity_costs_AU_Red_Energy_TOU.yaml

**Reply by SKNBP** [reply]
I have SUN2000-6KTL-M1 with DTSU666-H all 3-pahse, and you hack-s integration. All have been recognized without problems, even sensors for energy dashboard where there right out-of-the-box(i'm using those)

**Reply by SKNBP** [reply]
oh, my pv string is 6 kWp
